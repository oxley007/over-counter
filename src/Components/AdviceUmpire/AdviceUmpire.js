import React, { Component } from 'react';

class AdviceUmpire extends Component {
  constructor(props) {
    super(props);

    this.umpireAdvice = this.umpireAdvice.bind(this);
  }

umpireAdvice() {

  let randomAdive = Math.floor(Math.random()*7);
  let randomBattingBowling = Math.floor(Math.random()*7);
  if ((this.props.associated === 'batting' || this.props.associated === 'bowling') && randomBattingBowling < 2) {
    if (this.props.associated === 'batting')
    {
      let randomBowling = Math.floor(Math.random()*2);
      if (randomBowling === 0)
      {
        randomAdive = 7;
      }
      else {
        randomAdive = 8;
    }
  }
}
    if (this.props.associated === 'bowling')
    {
      let randomBowling = Math.floor(Math.random()*2);
      if (randomBowling === 0)
      {
        randomAdive = 9;
      }
      else {
        randomAdive = 10;
      }
    }
  console.log(randomAdive);
  switch (randomAdive) {
  case 0:
    return <p>LBW advice: watch the next few balls closely and take note if the ball is bouncing over the hight of the stumps</p>
    break;
  case 1:
    return <p>LBW advice: take note how far or how close the batter is standing to the stumps. </p>
    break;
  case 2:
    return <p>LBW advice: take note of the batters height. Where do the knee roles match up with the stumps?</p>
    break;
  case 3:
    return <p>No ball advice: are you watching the front foot?</p>
    break;
  case 4:
    return <p>Wide advice: do you have a good indication of where the off-side and leg-side wide markers are?</p>
    break;
  case 5:
    return <p>One-short advice: make sure you watch the batsman when running between the wickets to make sure they pass the crease.</p>
    break;
  case 6:
    return <p>Runout advice: are you moving square of the wickets for tight runs to best judge a runout?</p>
    break;
  case 7:
    return <p>Running advice: is the non-striker backing-up when the ball is bowled?</p>
    break;
  case 8:
    return <p>Running advice: are the batsman calling loudly. This includes “waiting” to see if the fielder will get to the ball. </p>
    break;
  case 9:
    return <p>Fielding advice: are the fielders backing up the wicket keeper and bowlers end to stop any over-throws?</p>
    break;
  case 10:
    return <p>Fielding advice: are the fielders walking-in with the bowler?</p>
    break;
  default:
    return <p>No ball advice: are you watching the front foot?</p>
    break;
}
}

  render() {
    return (
      <div>
      {this.umpireAdvice()}
      </div>
    );
  }
}

export default AdviceUmpire;
