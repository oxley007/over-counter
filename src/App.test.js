import React from 'react';
import ReactDOM from 'react-dom';
import AdviceUmpire from './Components/AdviceUmpire/AdviceUmpire';
import renderer from 'react-test-renderer';

/*
Need to add test the check the assert the output of 'batting' 'bowling' and 'neautral' from adviceumpire
*/

//to do

/*
Test Components*/

describe('AdviceUmpire component renders the AdviceUmpire correctly', () => {
  it('renders correctly', () => {
    const adviceumpire = { associatedWith: "batting"};
    const rendered = renderer.create(
      <AdviceUmpire associated={adviceumpire} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});

/*it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Add />, div);
  ReactDOM.unmountComponentAtNode(div);
});*/

describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  });
});
