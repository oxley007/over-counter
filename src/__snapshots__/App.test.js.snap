// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdviceUmpire component renders the AdviceUmpire correctly renders correctly 1`] = `
<div>
  <p>
    LBW advice: take note how far or how close the batter is standing to the stumps. 
  </p>
</div>
`;
